dumper/app/src/test/java/com/google/edwmigration/dumper/application/dumper/MetadataDumperTest.java

```diff
@@ -25,10 +32,108 @@
 @RunWith(JUnit4.class)
 public class MetadataDumperTest {
 
+  // TODO(ishmum): `testOverridesZipWithGivenName` with content check
+  private File file;
+  private MetadataDumper dumper = new MetadataDumper().withExitOnError(false);
+  private final Connector connector = new HiveMetadataConnector();
+  private final String defaultFileName = "dwh-migration-hiveql-metadata.zip";
+
+  @After
+  public void tearDown() throws IOException {
+    if (file == null || !file.exists()) return;
+    if (file.isDirectory()) FileUtils.deleteDirectory(file);
+    else file.delete();
+  }
+
   @Test
   public void testInstantiate() throws Exception {
-    MetadataDumper dumper = new MetadataDumper();
-    // dumper.run("--help");    // Calls System.exit(1)
+    dumper = new MetadataDumper();
     dumper.run("--connector", new BigQueryLogsConnector().getName(), "--dry-run");
   }
+
+  @Test
+  public void testCreatesDefaultOutputZip() throws Exception {
+    // Arrange
+    file = new File(defaultFileName);
+
+    // Act
+    dumper.run("--connector", connector.getName());
+
+    // Assert
+    Assert.assertTrue(file.exists());
```

Comment:
Since we're using Google style for this repo, I imagine (1) static imports are preferred here, and (2) we should be using https://github.com/google/truth instead of JUnit assertions.

Reply:
resolved in conversation